File: attendance2/solutions/attendance-zf.py
#!/bin/python3

N = int(input())

calls = [input() for _ in range(N)]

absent = []
for i in range(N):
  if calls[i] != "Present!" and (i == N-1 or calls[i+1] != "Present!"):
    absent.append(calls[i])

if absent: print(*absent, sep="\n")
else: print("No Absences")

File: attendance2/solutions/attendance-zf.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
  int N;
  cin >> N;

  string prev = "";
  int absent = 0;

  for (int i = 0; i < N; ++i) {
    string call;
    cin >> call;
    if (call == "Present!") prev = "";
    else {
      if (prev != "") {
        cout << prev << endl;
        ++absent;
      }
      prev = call;
    }
  }
  if (prev != "") {
    cout << prev << endl;
    ++absent;
  }
  
  if (absent == 0) cout << "No Absences" << endl;

  return 0;
}